packages/server/assets/core/typedefs/modelsAndVersions.graphql

```diff
@@ -0,0 +1,180 @@
+  # The id of the actual object containing the data from this version.
+  referencedObject: String!
+  # The name of the source application that this version originated from.
+  sourceApplication: String!
+  # The name of the file that this version originated from. NOTE: needs schema extending!
+  sourceFileName: String
+  # If set by the host application connector, the id of the previous version.
+  parents: [String]
+  author: LimitedUser
+}
+
+type ModelCollection {
+  totalCount: Int!
+  cursor: String
+  items: [Model!]
+}
+
+input ModelQuery {
+  name: String
+  # TODO: will be expanded
+}
+
+type VersionCollection {
+  totalCount: Int!
+  cursor: String
+  items: [Version!]
+}
+
+input ModelVersionQuery {
+  # text search by the version's message
+  message: String
+  # search by author
+  authorId: String
+  # search by source application
+  sourceApplication: String
+}
+
+extend type Mutation {
+  # Models C(R)UD
+  modelCreate(model: ModelCreateInput!): String!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+
+  modelUpdate(model: ModelUpdateInput!): Boolean!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+
+  modelDelete(model: ModelDeleteInput!): Boolean!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+
+  # Versions C(R)UD
+  versionCreate(version: VersionCreateInput!): String!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+
+  versionUpdate(version: VersionUpdateInput!): Boolean!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+
+  versionReceive(input: VersionReceivedInput!): Boolean!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:read")
+
+  versionDelete(version: VersionDeleteInput!): Boolean!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+
+  """
+  Move a batch of versions to a new branch
+  """
+  versionsMove(input: VersionsMoveInput!): Boolean!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+
+  """
+  Delete a batch of versions
+  """
+  versionsDelete(input: VersionsDeleteInput!): Boolean!
+    @hasRole(role: "server:user")
+    @hasScope(scope: "streams:write")
+}
+
+input ModelCreateInput {
+  projectId: String!
+  name: String!
+  description: String
+}
+
+input ModelUpdateInput {
+  projectId: String!
+  id: String!
+  name: String
+  description: String
+}
+
+input ModelDeleteInput {
+  projectId: String!
+  id: String!
+}
+
+input VersionCreateInput {
+  projectId: String!
+  modelId: String! # Note: we're no longer using branch names -> swithcing to pure model ids.
+  objectId: String!
+  message: String
+  sourceApplication: String
+  sourceFileName: String # Note: db schema extension needed
+  totalChildrenCount: Int
+  parents: [String]
+}
+
+input VersionUpdateInput {
+  projectId: String!
+  id: String!
+  message: String
+  """
+  To move the commit to a different model, please the id of the model.
```

Comment:
> please the id of the model
😳

Reply:
you choose, based on your preferences (and the model's)